pkg/driver/schd_helper.go

```diff
@@ -56,12 +60,44 @@ func getVolumeWeightedMap(pool string) (map[string]int64, error) {
 	return nmap, nil
 }
 
+// getCapacityWeightedMap goes through all the pools on all the nodes
+// and creates the node mapping of the capacity for all the nodes.
+// It returns a map which has nodes as key and capacity provisioned
+// on the nodes as corresponding value. The scheduler will use this map
+// and picks the node which is less weighted.
+func getCapacityWeightedMap(pool string) (map[string]int64, error) {
+	nmap := map[string]int64{}
+
+	zvlist, err := volbuilder.NewKubeclient().
+		WithNamespace(zfs.OpenEBSNamespace).
+		List(metav1.ListOptions{})
+
+	if err != nil {
+		return nmap, err
+	}
+
+	// create the map of the volume capacity
+	// for the given pool
+	for _, zv := range zvlist.Items {
+		if zv.Spec.PoolName == pool {
+			volsize, err := strconv.ParseInt(zv.Spec.Capacity, 10, 64)
```

Comment:
Is this going to get the actual used capacity or reserved capacity? 

For example, if the volumes are provisioned as follows on two nodes with 100GB pools:
- v1 = 50G => node 1 (this volume writes only 5mb of data)
- v2 = 50G => node 2 (this volume writes 1GB of data)

- v3 = 50G
- v4 = 50G

Will v3 and v4 go to node1 because it has occupied less capacity then node2?

Reply:
it is based on pv size. So ,in the above case v3 and v4 will go to different nodes (node1 and node2).

Reply:
Kiran, for Thick provisioned volumes does the actual space usage even matters.

Reply:
nope. for thick provisioned, it doesn't matter. using based on PV size is good.